parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriterV2.java

```diff
@@ -64,15 +65,17 @@ public ColumnWriterV2(
       ColumnDescriptor path,
       PageWriter pageWriter,
       ParquetProperties parquetProps,
-      int pageSize) {
+      int pageSize,
+      ByteBufferAllocator allocator) {
     this.path = path;
     this.pageWriter = pageWriter;
     resetStatistics();
 
-    this.repetitionLevelColumn = new RunLengthBitPackingHybridEncoder(getWidthFromMaxInt(path.getMaxRepetitionLevel()), MIN_SLAB_SIZE, pageSize);
-    this.definitionLevelColumn = new RunLengthBitPackingHybridEncoder(getWidthFromMaxInt(path.getMaxDefinitionLevel()), MIN_SLAB_SIZE, pageSize);
+    this.repetitionLevelColumn = new RunLengthBitPackingHybridEncoder(getWidthFromMaxInt(path.getMaxRepetitionLevel()), MIN_SLAB_SIZE, pageSize, allocator);
+    this.definitionLevelColumn = new RunLengthBitPackingHybridEncoder(getWidthFromMaxInt(path.getMaxDefinitionLevel()), MIN_SLAB_SIZE, pageSize, allocator);
 
     int initialSlabSize = CapacityByteArrayOutputStream.initialSlabSizeHeuristic(MIN_SLAB_SIZE, pageSize, 10);
+    // TODO - Should I be passing an allocator here rather than just letting it use the one stored in ParquetProperties
```

Comment:
Removed cases where an argument list contained both an allocator and a ParquetProperties object. In  InternalParquetRecordWriter.java where we were previously keeping separate references to an allocator and a Properties object I removed the local reference to the allocator and just grab the one out of the properties object for everything.